packages/ruleset/src/functions/pattern-properties.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2017 - 2023 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const {
+  isObject,
+  validateNestedSchemas,
+} = require('@ibm-cloud/openapi-ruleset-utilities');
+const { LoggerFactory } = require('../utils');
+
+let ruleId;
+let logger;
+
+module.exports = function (schema, _opts, context) {
+  if (!logger) {
+    ruleId = context.rule.name;
+    logger = LoggerFactory.getInstance().getLogger(ruleId);
+  }
+  return validateNestedSchemas(
+    schema,
+    context.path,
+    patternPropertiesCheck,
+    true,
+    false
+  );
+};
+
+/**
+ * Enforces certain restrictions on the use of "patternProperties" within a schema:
+ * 1. patternProperties and additionalProperties are mutually exclusive
+ * 2. patternProperties must be an object
+ * 3. patternProperties must not be empty
+ * 4. patternProperties must have at most one entry
+ * 5. values in patternProperties entry must be a schema object (this check is actually
+ *    not implemented here since our validateNestedSchemas function will detect a non-object
+ *    where a schema should appear)
```

Comment:
We can probably remove the bullet if we aren't going to implement it here, and I think it's valid to leave it out of the scope for this rule.